internal/provider/resource_work_space_alert.go

```diff
@@ -0,0 +1,361 @@
+			resp.Diagnostics.AddError("Error determining notify_on value", err.Error())
+			return
+		}
+
+		alertingRule := workspaceAlert{
+			Channel:          types.StringValue(rule.Channels[0].ID),
+			PreviewImage:     types.BoolValue(rule.Channels[0].IncludePreviewImage),
+			NotifyOn:         types.StringValue(notifyOn),
+			SelectedMonitors: selectedMonitors,
+		}
+		alertingRules = append(alertingRules, alertingRule)
+	}
+
+	updatedState := workspaceAlerts{
+		WorkspaceID:   state.ID,
+		AlertingRules: alertingRules,
+		ID:            state.ID,
+	}
+
+	diags = resp.State.Set(ctx, &updatedState)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (r *workspaceAlertResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var plan workspaceAlerts
+	diags := req.Plan.Get(ctx, &plan)
+	if diags.HasError() {
+		resp.Diagnostics = diags
+		return
+	}
+
+	payload, err, warning := constructPayload(plan)
+	if err != nil {
+		resp.Diagnostics.AddError("Error constructing JSON", err.Error())
+		return
+	}
+
+	if warning != "" {
+		resp.Diagnostics.AddWarning("Unsupported Attribute", warning)
+	}
+
+	err = r.client.UpdateWorkspace(plan.WorkspaceID.ValueString(), payload)
+	if err != nil {
+		resp.Diagnostics.AddError("Error updating workspace alerts", err.Error())
+		return
+	}
+
+	plan.ID = types.StringValue(plan.WorkspaceID.ValueString())
+
+	diags = resp.State.Set(ctx, plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (r *workspaceAlertResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state workspaceAlerts
+	diags := req.State.Get(ctx, &state)
+	if diags.HasError() {
+		resp.Diagnostics = diags
+		return
+	}
+
+	payload := map[string]interface{}{
+		"alertingRules": []interface{}{},
+	}
+
+	err := r.client.UpdateWorkspace(state.ID.ValueString(), payload)
+	if err != nil {
+		resp.Diagnostics.AddError("Error with removing workspace alerts", err.Error())
+		return
+	}
+}
+
+func (r *workspaceAlertResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+func constructPayload(plan workspaceAlerts) (map[string]interface{}, error, string) {
+	var result WorkspaceAlertsData
+	var warning string
+
+	for _, rule := range plan.AlertingRules {
+		var channels []AlertChannel
+
+		channel := AlertChannel{
+			ID:                  rule.Channel.ValueString(),
+			IncludePreviewImage: rule.PreviewImage.ValueBool(),
+		}
+
+		if rule.NotifyOn.ValueString() == "workspace_state" && rule.PreviewImage.ValueBool() {
+			channel.IncludePreviewImage = false
+			warning = "Preview images are not supported when using 'workspace_state' for 'notify_on'. The 'preview_image' attribute will be ignored."
+		}
+
+		channels = append(channels, channel)
```

Comment:
It looks like we're only supporting specifying a single channel in the alertingRules channels array?